lib/librato/collector/aggregator.rb

```diff
@@ -94,17 +95,21 @@ def measure(*args, &block)
         end
 
         percentiles = Array(options[:percentile])
-        tags_payload =
-          if options[:source]
-            # convert custom instrumentation using legacy source
-            { tags: { source: options[:source] } }
-          elsif options[:tags]
-            { tags: options[:tags] }
+        source = options[:source]
+        tags_option = options[:tags]
+        tags_option = { source: source } if source && !tags_option
+        tags =
+          if @default_tags && tags_option && options[:inherit_tags]
+            @default_tags.merge(tags_option)
+          elsif tags_option
+            tags_option
+          else
+            @default_tags
```

Comment:
I think this logic could be simplified a bit if `@default_tags` was always a hash. You can argue for the optimization of skipping the merge if default tags is not set but realistically it almost always will be.